dds_web/api/files.py

```diff
@@ -69,6 +70,7 @@ class NewFile(flask_restful.Resource):
     """Inserts a file into the database"""
 
     @auth.login_required(role=["Super Admin", "Unit Admin", "Unit Personnel"])
+    @logging_bind_request
```

Comment:
We definitely need logging here too but I'm wondering about the `flask.request.json` -- in this (and a few more) case the json info will include all info that will be put into the database. So we should maybe have a check and for some endpoints not log that.

Reply:
I will have a look!

Reply:
`dds_cli/dev` works with this branch, but you're probably getting the bucket error because you're using the development entries - the projects that are automatically created when starting up the service. So do (in dds_cli):
```
git checkout dev
git pull
dds project create -t test -d test -pi test
dds data put -p <new-project-id-displayed-from-create> -s <file-you-want-to-upload>
```

When I do this the action logger outputs the following
```
dds_backend | [2022-01-19 07:40:06,260] _base [INFO] {"resource": "/api/v1/file/new", "request_args": {"name": "requirements.txt", "name_in_bucket": "0fe9fe9c-1e57-5095-8f94-9b7102c41ae2", "subpath": ".", "size": 267, "size_processed": 250, "compressed": true, "salt": "B639892A751B4ABC94E33FCAAB6D9972", "public_key": "E091222CBE0E063E00F811C015ABE9F383C61A578DE12A5DD5351B596CEF4B55", "checksum": "e449dba65a98df64fc7eedb4ed1d49dc2cc58d89f6098ae8783cb284d0d16362"}, "user": "unituser_1", "response": "200 OK", "event": "api_blueprint.new_file.post", "logger": "actions", "level": "info", "timestamp": "2022-01-19T07:40:06.260174Z"}
```

Reply:
@MatthiasZepper 
We could have something like this:
```python
# dds_web/api/dds_decorators.py

def get_user_to_log():
    if auth.current_user():
        current_user = auth.current_user().username
    elif flask_login.current_user.is_authenticated:
        current_user = flask_login.current_user.username
    elif flask.request.remote_addr:
        current_user = flask.request.remote_addr  # log IP instead of username
    elif flask.request.access_route:
        current_user = flask.request.access_route[0]  # log IP instead of username
    else:
        current_user = "anonymous"
    return current_user


def do_logging(log_json=False):
    def logging_decorator_name(func):
        """Binds some request parameters to the thread-local context of structlog"""

        @functools.wraps(func)
        def wrapper_logging_bind_request(*args, **kwargs):

            flask.current_app.logger.debug(f"Test: {log_json}")
            current_user = get_user_to_log()

            with structlog.threadlocal.bound_threadlocal(
                resource=flask.request.path or "not applicable",
                request_args=flask.request.json if (flask.request.data and log_json) else None,
                user=current_user,
            ):

                value = func(*args, **kwargs)

                if hasattr(value, "status"):
                    structlog.threadlocal.bind_threadlocal(response=value.status)

                action_logger.info(f"{flask.request.endpoint}.{func.__name__}")

                return value

        return wrapper_logging_bind_request

    return logging_decorator_name
```
And then at the endpoints we do want to log json we do (the `log_json` could also of course be `True` by default if we decide that is better)
```
    @auth.login_required(role=["Super Admin", "Unit Admin", "Unit Personnel"])
    @do_logging()
    def post(self):
```

Reply:
> ```shell
> dds project create -t test -d test -pi test
> dds data put -p <new-project-id-displayed-from-create> -s <file-you-want-to-upload>
> ```

YES, thank you! I always tried uploading into one of the existing projects! Now I finally know how that works!

> ```shell
>  @do_logging(log_json=True)
> ```

Cool, decorators with arguments. That is impressive!

However, I would suggest a different approach: Instead of logging everything or nothing, I have now created a utility function that will censor sensitive (or irrelevant) arguments from the requests prior to logging them. I have put the droplist/blocklist into the config, which to me seemed a better place than to hardcode it inside the function. 

Admittedly, it lacks the brevity of the other config entries (thus clutters the config) and will probably hardly change, so it might not be necessary to have it in the config?

Reply:
My reply: https://github.com/ScilifelabDataCentre/dds_web/pull/771#discussion_r788402501